tests/playwright/src/utility/cluster-operations.ts

```diff
@@ -0,0 +1,171 @@
+      timeout: 15_000,
+    });
+    await statusBar.validateKubernetesContext(`kind-${clusterName}`);
+  });
+}
+
+export async function deleteCluster(
+  page: Page,
+  resourceName: string = 'kind',
+  containerName: string = 'kind-cluster-control-plane',
+  clusterName: string = 'kind-cluster',
+): Promise<void> {
+  return test.step('Delete Kind cluster', async () => {
+    const navigationBar = new NavigationBar(page);
+    const resourceCard = new ResourceConnectionCardPage(page, resourceName, clusterName);
+    const volumeName = await getVolumeNameForContainer(page, containerName);
+
+    await navigationBar.openSettings();
+    const resourcesPage = new ResourcesPage(page);
+    await playExpect(resourcesPage.heading).toBeVisible({ timeout: 10_000 });
+    if (!(await resourceCard.doesResourceElementExist())) {
+      console.log(`Kind cluster [${clusterName}] not present, skipping deletion.`);
+      return;
+    }
+
+    await resourceCard.performConnectionAction(ResourceElementActions.Stop);
+    await playExpect(resourceCard.resourceElementConnectionStatus).toHaveText(ResourceElementState.Off, {
+      timeout: 50_000,
+    });
+    await resourceCard.performConnectionAction(ResourceElementActions.Delete);
+    await playExpect(resourceCard.markdownContent).toBeVisible({
+      timeout: 50_000,
+    });
+    const containersPage = await navigationBar.openContainers();
+    await playExpect(containersPage.heading).toBeVisible();
+    await playExpect
+      .poll(async () => containersPage.containerExists(containerName), {
+        timeout: 10_000,
+      })
+      .toBeFalsy();
+
+    const volumePage = await navigationBar.openVolumes();
+    await playExpect(volumePage.heading).toBeVisible();
+    await playExpect
+      .poll(async () => await volumePage.waitForVolumeDelete(volumeName), {
+        timeout: 20_000,
+      })
+      .toBeTruthy();
+  });
+}
+
+export async function checkClusterResources(page: Page, containerName: string): Promise<void> {
+  return test.step('Check resources added with cluster', async () => {
+    const navigationBar = new NavigationBar(page);
+    const containersPage = await navigationBar.openContainers();
+    await playExpect.poll(async () => containersPage.containerExists(containerName)).toBeTruthy();
+    const containerDetailsPage = await containersPage.openContainersDetails(containerName);
+    await playExpect.poll(async () => await containerDetailsPage.getState()).toEqual(ContainerState.Running);
+
+    const volumesPage = new VolumesPage(page);
+    const volumeName = await getVolumeNameForContainer(page, containerName);
+    if (!volumeName) {
+      throw new Error(`Volume name for container "${containerName}" is not defined.`);
+    }
+    const volumeDetailsPage = await volumesPage.openVolumeDetails(volumeName);
+    await playExpect.poll(async () => await volumeDetailsPage.isUsed()).toBeTruthy();
+  });
+}
+
+export async function clusterOperations(
+  page: Page,
```

Comment:
please change to perform one particular action, ie defined by ResourceElementAction (maybe change name to resourceConnectionActions?) and check for expected state.